pkg/db/migrations/migrations.go

```diff
@@ -93,18 +55,66 @@ func migrate(ctx context.Context, db *sql.DB, list []string, assets http.FileSys
 			continue
 		}
 
-		res, err := tx.ExecContext(ctx, stmt)
-		if err != nil {
-			logger.WithError(err).Error("migration failed")
-			return errors.Wrap(err, "migration failed")
-		}
-		affected, err := res.RowsAffected()
+		err = runStatement(ctx, db, stmt)
```

Comment:
@trusch this moves each statement into its own transaction, but there is a major flaw if we do this: if one of the migrations fail, there is no way to rollback the previous migrations that were applied in this release process.  I am planning on reverting this part of the change. 

I have checked a few other migration systems, for example https://github.com/jackc/tern/blob/5f725f9e1b8aeb27204a47c95ff01de34eec70f2/migrate/migrate.go
which first gathers all of the required sql statements into a single sql string, and then runs that aggregated sql in a transaction. I checked a few other systems, and it was similar. All of the migrations _need_ to be run in a single transaction to ensure that you can fail gracefully.

Reply:
@pragmader I'd like to get your opinion on this. I like it that way because it isolates the migration statements, but we can end up in a half migrated state if only a bunch of statements were successfull